UI/output-obj.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "obs.hpp"
+
+enum class OutputType {
+	Unknown,
+	Preview,
+	Program,
+	Source,
+	Multiview, // TODO
+};
+
+class OutputObj {
+	bool active = false;
+
+	enum OutputType type;
+
+	OBSOutputAutoRelease output;
+	OBSWeakSourceAutoRelease weakSource;
+
+	obs_view_t *view = nullptr;
+	video_t *video = nullptr;
+
+public:
+	OBSOutput GetOutput();
+
+	enum OutputType GetType();
+	void SetType(enum OutputType type);
+
+	bool Active();
+
+	bool Start();
+	void Stop();
+
+	void SetSource(OBSSource source);
+	OBSSource GetSource();
```

Comment:
Returning an RAII is technically not necessary, as it'd theoretically increment the reference then immediately decrement it again. `obs_source_t *` should work fine

Reply:
@Lain-B has stated before that RAII should always be used with cpp.